CMake/External_SWIG.cmake

```diff
@@ -0,0 +1,43 @@
+
+if(WIN32)
+  ExternalProject_Add(SWIG
+    URL ${SWIG_url}
+    URL_MD5 ${SWIG_md5}
+    PREFIX ${fletch_BUILD_PREFIX}
+    DOWNLOAD_DIR ${fletch_DOWNLOAD_DIR}
+    INSTALL_DIR ${fletch_BUILD_INSTALL_PREFIX}
+    CMAKE_GENERATOR ${gen}
+    CMAKE_ARGS
+      ${COMMON_CMAKE_ARGS}
+      -DCMAKE_CXX_COMPILER:FILEPATH=${CMAKE_CXX_COMPILER}
+      -DCMAKE_C_COMPILER:FILEPATH=${CMAKE_C_COMPILER}
+      -DPYTHON_EXECUTABLE=${PYTHON_EXECUTABLE}
+      -DPYTHON_INCLUDE_DIR=${PYTHON_INCLUDE_DIR}
+      -DPYTHON_LIBRARY=${PYTHON_LIBRARY}
+      -DBUILD_SHARED_LIBS:BOOL=ON
+    )
+else()
+  ExternalProject_Add(SWIG
+    URL ${SWIG_url}
+    URL_MD5 ${SWIG_md5}
+    PREFIX ${fletch_BUILD_PREFIX}
+    DOWNLOAD_DIR ${fletch_DOWNLOAD_DIR}
+    INSTALL_DIR ${fletch_BUILD_INSTALL_PREFIX}
+    BUILD_IN_SOURCE 1
+    CONFIGURE_COMMAND ./autogen.sh && ./configure
```

Comment:
Even if you insist `make` is the only way to build on non-win32, there is a macro which specifies this project uses make. You can find that example in another External_* file.